cookbook/integrations/gcp/bigquery/bigquery.py

```diff
@@ -0,0 +1,49 @@
+"""
+BigQuery Query
+############
+
+This example shows how to use a Flyte BigQueryTask to execute a query.
+"""
+
+from flytekit import kwtypes, workflow
+from flytekitplugins.bigquery import BigQueryConfig, BigQueryTask
+# %%
+# This is the world's simplest query. Note that in order for registration to work properly, you'll need to give your
+# BigQuery task a name that's unique across your project/domain for your Flyte installation.
+bigquery_task_no_io = BigQueryTask(
+    name="sql.bigquery.no_io",
+    inputs={},
+    query_template="SELECT 1",
+    output_schema_type=None,
+    task_config=BigQueryConfig(ProjectID="flyte", Location="us-west1-b"),
+)
+
+
+@workflow
+def no_io_wf():
+    return bigquery_task_no_io()
+
+
+# %%
+# Of course, in real world applications we are usually more interested in using BigQuery to query a dataset.
+# In this case we use crypto_dogecoin data which is public dataset in BigQuery.
+# `here <https://console.cloud.google.com/bigquery?project=bigquery-public-data&page=table&d=crypto_dogecoin&p=bigquery-public-data&t=transactions>`__
+#
+# Let's look out how we can parameterize our query to filter results for a specific transaction version, provided as a user input
+# specifying a version.
+
+bigquery_task_templatized_query = BigQueryTask(
+    name="sql.bigquery.w_io",
+    # Define inputs as well as their types that can be used to customize the query.
+    inputs=kwtypes(version=int),
+    task_config=BigQueryConfig(ProjectID="flyte", Location="us-west1-b"),
+    query_template="SELECT * FROM `bigquery-public-data.crypto_dogecoin.transactions` WHERE @version = 1 LIMIT 10;",
+)
+
+
+@workflow
+def full_bigquery_wf(version: int):
+    return bigquery_task_templatized_query(version=version)
```

Comment:
What is the output type? I think it'll be interesting to demonstrate how to use the out put too.

Reply:
The type will be `StructuredDataset`. it's a new type introduced after flytekit `v0.30.0b3` and is a superset of flyte schema.
It can transparently read different data sources from (BigQuery, S3,...) and convert result tables or parquet file to `pd.Dataframe` or `pyarrow.Table`.

Reply:
for more detail, you can check https://github.com/flyteorg/flytekit/pull/785

Reply:
@pradithya so we were going to use Flyte schema, but we have been working on a better representation for structureddatasets in general- dataframes, tensors